src/chromo/common.py

```diff
@@ -721,8 +750,10 @@ def set_stable(self, pdgid, stable=True):
         if abs(pdgid) == 311:
             self._set_stable(130, stable)
             self._set_stable(310, stable)
+            self._update_final_state_particles([130, 310], stable)
```

Comment:
Here, that's what I mean, you are duplicating functionality. Remove `_update_final_state_particles` and integrate any code that you need from there into `set_stable`.